### PR TITLE
fix(review-loop): require substantive settle reviews

### PR DESCRIPTION
**Pull request by lhpaul**
## Summary
- require post-clean submitted-review probes to ignore empty reply-container review objects
- bind review evidence to the current head when GitHub exposes review commit_id
- add focused regression coverage for empty containers, substantive reviews, mixed evidence, and other-head containers

## Pre-Submission Self-Review
- Scope check: diff is limited to pr-review-loop settle review detection, its focused shell harness tests, and CHANGELOG.
- Issue coverage: addresses #1596 AC-1 through AC-4 by centralizing substantive-body classification, preserving POST_CLEAN_NO_SUBMITTED_REVIEW behavior when only containers exist, and adding regression fixtures based on GitHub review-object body shape.
- Risk: medium workflow-tooling change; no schema, external configuration, or unrelated reviewer platforms changed.

## Validation
- bash -n scripts/development-workflow/pr-review-loop.sh
- bash -n scripts/development-workflow/tests/test-pr-review-loop.sh
- bash scripts/development-workflow/tests/test-pr-review-loop.sh --area 19 (52 passed, 0 failed)
- shellcheck --severity=warning scripts/development-workflow/pr-review-loop.sh scripts/development-workflow/tests/test-pr-review-loop.sh
- python3 scripts/lint/workflow-shell-guard-lint.py --base-ref origin/develop
- bash scripts/lint/check-changelog-duplicate-headers.sh CHANGELOG.md
- git diff --check

Closes #1596

---

> [!NOTE]
> **Medium Risk**
> Changes when the review loop declares a head “reviewed” for settle, which can lengthen waits or alter automation timing if bots only emit empty review containers.
> 
> **Overview**
> Fixes **post-clean settle** so `POST_CLEAN_REQUIRE_REVIEW` no longer treats GitHub’s empty “reply-container” review objects as a completed bot review on the current head.
> 
> `_bot_review_submitted_since` now takes an optional **head SHA**, ignores reviews whose `commit_id` points at another commit (while still accepting missing `commit_id`), and only returns success when at least one matching submitted review has a **substantive body** via new `_review_body_is_substantive` (non-empty text, excluding short acknowledgments like “Thanks!”). The settle loop passes **`$settle_head_sha`** into that probe. Regression tests cover empty containers, head binding, mixed evidence, and stale/other-head cases; **CHANGELOG** documents the fix for #1596.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit db8d15b9df54f0578bebb88ac69cf61b56a2a89a. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by lhpaul** [comment]
### Step 7a Internal Review Gate Summary

Reviewer set: codex (reachable from current Codex runner).

Pass 1: APPROVED — implementation covers #1596: empty review-container objects no longer satisfy the submitted-review settle, substantive review evidence still settles, and current-head binding is enforced when review commit_id is available.

Pass 2: APPROVED after direct fix — PR-Agent identified acknowledgement-only punctuation such as "Thanks!" as an edge case; fixed in commit 738c12eb and covered by the focused harness.

Validation observed:
- bash -n scripts/development-workflow/pr-review-loop.sh
- bash -n scripts/development-workflow/tests/test-pr-review-loop.sh
- bash scripts/development-workflow/tests/test-pr-review-loop.sh --area 19 (53 passed, 0 failed)
- shellcheck --severity=warning scripts/development-workflow/pr-review-loop.sh scripts/development-workflow/tests/test-pr-review-loop.sh
- python3 scripts/lint/workflow-shell-guard-lint.py --base-ref origin/develop
- git diff --check

Verdict: APPROVED

**Comment by lhpaul** [comment]
bugbot run

**Comment by lhpaul** [comment]
### Automated Reviewer Loop Summary

**Result:** clean — no blocking findings
**Platforms:** pr-agent (clean), bugbot (clean)
**Findings:** 0 blocking, 0 suggestions

**Ready reviewer phase:** No net-new blocker was found after the draft GitHub gate.
**Ready-phase platforms:** bugbot

**Advisory findings (non-blocking):**
- Possible Issue ([view comment](https://github.com/lhpaul/ai-dev-framework-template/pull/1611#issuecomment-5424363630))

*Posted automatically by `pr-review-loop.sh`.*

<details>
<summary>Reviewer-loop history (2 iterations)</summary>


```json
{
  "schema": "reviewer_loop_history.v1",
  "pr_number": 1611,
  "updated_at": "2026-08-26T11:21:18Z",
  "history_status": "available",
  "history_unavailable_reason": "",
  "entries": [
    {
      "iteration": 1,
      "recorded_at": "2026-08-26T11:12:58Z",
      "head_sha": "738c12eb9f31b2cd1f1f8d12b90c5936f03a146e",
      "run_id": "auto-1787742461-98317-13266",
      "result": "needs_fixes",
      "reason": "unresolved_review_threads",
      "platforms": [
        "pr-agent (clean)",
        "bugbot (clean)"
      ],
      "blocking_count": 1,
      "suggestion_count": 0,
      "unresolved_thread_count": 1,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "bugbot"
        ],
        "started": true,
        "net_new_blocker": true,
        "blocking_platform": "review_threads"
      }
    },
    {
      "iteration": 2,
      "recorded_at": "2026-08-26T11:21:18Z",
      "head_sha": "db8d15b9df54f0578bebb88ac69cf61b56a2a89a",
      "run_id": "auto-1787743048-59359-30086",
      "result": "clean",
      "reason": "",
      "platforms": [
        "pr-agent (clean)",
        "bugbot (clean)"
      ],
      "blocking_count": 0,
      "suggestion_count": 0,
      "unresolved_thread_count": 0,
      "late_threads_found": 0,
      "phase_after_clean": {
        "enabled": true,
        "platforms": [
          "bugbot"
        ],
        "started": true,
        "net_new_blocker": false,
        "blocking_platform": ""
      }
    }
  ]
}
```
</details>

File: scripts/development-workflow/pr-review-loop.sh [hunk 1]
```diff
@@ -7663,20 +7684,45 @@ _settle_config_for_platform() {
...
             | ((($bots | index($l)) != null) or (($bots | index($raw)) != null))
-          ) | select((.submitted_at // "") > $since)
-        ] | length' 2>/dev/null)" || n=""
-  case "$n" in
-    ''|*[!0-9]*) printf '%s' "-1"; return 0 ;;
-  esac
-  [ "$n" -gt 0 ] && printf '1' || printf '0'
+          )
+        | select((.submitted_at // "") > $since)
+        | select(($head == "") or ((.commit_id // "") == $head))
```
[the remark this answers is not in the document]
Comment (lhpaul) [reply scripts/development-workflow/pr-review-loop.sh]: Fixed in db8d15b9. `_bot_review_submitted_since` now treats a missing `commit_id` as unbound while still rejecting an explicit different commit id; added `review_probe_detects_submitted_review_for_head_without_commit_id`. Validation: `bash scripts/development-workflow/tests/test-pr-review-loop.sh --area 19` (54 passed), shellcheck, guard lint, changelog duplicate-header check, and `git diff --check`.

**Comment by lhpaul** [comment]
bugbot run
